explorer/interpreter/value_kind.def

```diff
@@ -0,0 +1,61 @@
+// Part of the Carbon Language project, under the Apache License v2.0 with LLVM
+// Exceptions. See /LICENSE for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+// Note that this is an X-macro header.
+//
+// It does not use `#include` guards, and instead is designed to be `#include`ed
+// after the x-macro is defined in order for its inclusion to expand to the
+// desired output. The x-macro for this header is `CARBON_VALUE_KIND`. The
+// definition provided will be removed at the end of this file to clean up.
+
+#ifndef CARBON_VALUE_KIND
+#error "Must define the x-macro to use this file."
+#endif
+
+CARBON_VALUE_KIND(IntValue)
```

Comment:
And idea would be to change this to `CARBON_VALUE_KIND(IntValue, "an integer value")`, so the string is customizable and include the article. Would that address the concerns here?

Reply:
FWIW, my experience with Clang makes me skeptical about how well this will work long term. I think we'll need something fairly similar to Clang's formatted rendering if messages at least. You can see a fairly simple model for this in the Carbon toolchain/diagnostics library.

But while the primary users of the Carbon explorer are also very likely *developing* Carbon (in one form or another), it seems relatively reasonable to use something search-able in the codebase like the enum itself, and then work on building a more user-facing diagnostics system when we're expecting to have a more broad userbase. Right now, it feels a bit like we're investing time trying to build "good" user facing messages when that's not the most relevant thing.

But on the flip side, getting this exactly "right" also doesn't seem critical. I was mostly chiming in to try to help get *some* path forward here...